### PR TITLE
Feature/pc retailer registration

### PR DESCRIPTION
**Pull request by LazarJovic**

### CODE REVIEW

File: Bitcoin-Payment-Service/src/main/java/goveed20/BitcoinPaymentService/aspects/LoggingAspect.java [hunk 1]
```diff
@@ -19,12 +19,13 @@
...
 @Component
-public class PaymentAspect {
+public class LoggingAspect {
 
     @Autowired
     private AsyncLogging asyncLogging;
 
-    @Before("execution(public * goveed20.BitcoinPaymentService.services.PaymentService.*(..)) || " +
+    @Before("execution(public * goveed20.BitcoinPaymentService.services.PaymentService.*(..)) ||" +
+            "execution(public * goveed20.BitcoinPaymentService.services.RetailerService.*(..)) ||" +
```
Comment (MasterRadule) [inline Bitcoin-Payment-Service/src/main/java/goveed20/BitcoinPaymentService/aspects/LoggingAspect.java:28]: Add this in Card payment service

File: Bitcoin-Payment-Service/src/main/java/goveed20/BitcoinPaymentService/services/PaymentService.java [hunk 2]
```diff
@@ -95,4 +92,22 @@ public void sendTransactionResponse(Long transactionId, TransactionStatus status
...
+
+    public Set<RegistrationField> getPaymentServiceRegistrationFields() {
+        Map<String, String> validationConstraints = new HashMap<>();
+        validationConstraints.put("type", "text");
+        validationConstraints.put("required", "required");
+
+        RegistrationField coinGateApiKey = RegistrationField.builder()
+                .encrypted(true)
+                .name("coinGateApiKey")
+                .type(RegistrationFieldType.STRING)
```
Comment (MasterRadule) [inline Bitcoin-Payment-Service/src/main/java/goveed20/BitcoinPaymentService/services/PaymentService.java:104]: Delete field "type" from RegistrationField

File: payment-concentrator-frontend/.eslintcache [hunk 3]
```diff
@@ -0,0 +1 @@
...ment-concentrator-frontend\\src\\components\\Toaster.js",[],{"ruleId":"45","replacedBy":"46"},{"ruleId":"47","replacedBy":"48"},{"ruleId":"49","severity":1,"message":"50","line":4,"column":40,"nodeType":"51","messageId":"52","endLine":4,"endColumn":45},{"ruleId":"53","severity":1,"message":"54","line":55,"column":8,"nodeType":"55","endLine":55,"endColumn":27,"suggestions":"56"},{"ruleId":"49","severity":1,"message":"57","line":3,"column":7,"nodeType":"51","messageId":"52","endLine":3,"endColumn":18},"no-native-reassign",["58"],"no-negated-in-lhs",["59"],"no-unused-vars","'Toast' is defined but never used.","Identifier","unusedVar","react-hooks/exhaustive-deps","React Hook useEffect has missing dependencies: 'checkServices', 'encryptionFields', 'formFields', and 'state'. Either include them or remove the dependency array.","ArrayExpression",["60"],"'PC_BASE_URL' is assigned a value but never used.","no-global-assign","no-unsafe-negation",{"desc":"61","fix":"62"},"Update the dependencies array to be: [availableServices, checkServices, encryptionFields, formFields, state]",{"range":"63","text":"64"},[2179,2198],"[availableServices, checkServices, encryptionFields, formFields, state]"]
```
Comment (draganculibrk9) [inline payment-concentrator-frontend/.eslintcache:1]: This file should be added to .gitignore

File: payment-concentrator-frontend/src/services/paymentService.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import axios from 'axios';
+
+const PC_BASE_URL = process.env.PC_BASE_URL // TODO: It gives undefined
```
Comment (draganculibrk9) [inline payment-concentrator-frontend/src/services/paymentService.js:3]: Remove `TODO`

**Review by draganculibrk9 — CHANGES_REQUESTED** [review]
Minor changes required

**Review by MasterRadule — CHANGES_REQUESTED** [review]

**Review by draganculibrk9 — APPROVED** [review]

**Review by MasterRadule — APPROVED** [review]

**Pull request merged**